packages/web3-core/src/web3_context.ts

```diff
@@ -135,6 +135,15 @@ export class Web3Context<
 		return this._subscriptionManager;
 	}
 
+	protected setRegisteredSubscriptions(registeredSubscriptions: RegisteredSubs): void {
+		if (!this.subscriptionManager) {
+			this._subscriptionManager = new Web3SubscriptionManager(
+				this.requestManager,
+				registeredSubscriptions,
+			);
+		}
+	}
```

Comment:
The type `RegisteredSubs` is generic and needs to be passed in the class definition. So does not seems correct to  specify type (which is subscription classes) at definition and actual values later on. The type `RegisteredSubs` is generic and needs to be passed in the class definition. So does not seems correct to  specify type (which is subscription classes) at definition and actual values later on.